### PR TITLE
feat(vapor-runtime): object support in `v-for`

### PR DESCRIPTION
**Pull request by wangyewei**
add support for object iteration in v-for

### CODE REVIEW

**Review by sxzz — CHANGES_REQUESTED** [review]
Conflicts need to be resolved

File: packages/runtime-vapor/src/for.ts [hunk 1]
```diff
@@ -136,9 +149,23 @@ export const createFor = (
       } else if (!getKey) {
         // unkeyed fast path
         const commonLength = Math.min(newLength, oldLength)
-        for (let i = 0; i < commonLength; i++) {
-          update((newBlocks[i] = oldBlocks[i]), source[i])
+        if (isObject(source) && !isArray(source)) {
+          const keys = Object.keys(source)
```
Comment (sxzz) [inline packages/runtime-vapor/src/for.ts:153]: Can we reuse `keys`?

File: packages/runtime-vapor/src/for.ts [hunk 2]
```diff
@@ -55,9 +56,18 @@ export const createFor = (
   }
 
   const mountList = (source: any[], offset = 0) => {
```
Comment (sxzz) [inline packages/runtime-vapor/src/for.ts:58]: `source` could be Record

**Comment by sxzz** [comment]
Thanks for your contribution. Implemented via 4af8e879 which combined with your commit.

**Comment by septatrix** [comment]
I assume the readme should also be updated?

https://github.com/vuejs/core-vapor/blob/d3ca3f7492cbd685aef934acd8dadc8069829930/README.md?plain=1#L47-L49

**Comment by sxzz** [comment]
Thanks for your reminder. Updated!

**Pull request closed without merging**